libs/utils/analysis/frequency_analysis.py

```diff
@@ -163,6 +163,127 @@ def _dfg_cpu_frequency_transition_rate(self, cpu):
 # Plotting Methods
 ###############################################################################
 
+    def plotPeripheralClock(self, clk, title='Peripheral Frequency'):
+        """
+        Produce graph plotting the frequency of a particular peripheral clock
+
+        :param title: The title for the chart
+        :type  title: str
+
+        :param clk: The clk name to chart
+        :type  clk: str
+
+        :raises: KeyError
+        """
+        if not self._trace.hasEvents('clock_set_rate'):
+            self._log.warning('Events [clock_set_rate] not found, plot DISABLED!')
+            return
+        rate_df = self._dfg_trace_event('clock_set_rate')
+        enable_df = self._dfg_trace_event('clock_enable')
+        disable_df = self._dfg_trace_event('clock_disable')
+        pd.options.mode.chained_assignment = None
```

Comment:
Usually when we face SettingWithCopy warnings we try to quiet them down by using `df.(i)loc ...`, even if the df is being modified properly. 

I see that this is isn't followed in plotClusterFrequencies(), but the warning is eventually restored with `pd.options.mode.chained_assignment = 'warn'`. If we really can't change the code to not have the warning I'd be fine with temporarily disabling it, but then it should be restored eventually.

Reply:
Likelt plotClusterFrequencies is warning because the pandas library added that check after that function as been added... but, going forward, I would say we should try to avoid warnings instead of muting them.

Reply:
Changed to not suppress warning, and corrected offending line to not generate warning.